pkg/pdatautil/hash.go

```diff
@@ -83,20 +83,26 @@ func ValueHash(v pcommon.Value) [16]byte {
 }
 
 func (hw *hashWriter) writeMapHash(m pcommon.Map) {
-	hw.keysBuf = hw.keysBuf[:0]
+	nextIndex := len(hw.keysBuf)
```

Comment:
I was stuck on thinking that this wouldn't work because buffers are reused. However, I see that ultimately the buffer is wiped before returning to the pool. It only retains values so long as it's processing a map and its nested maps. Clever, but I think this warrants an explanatory comment.

Reply:
Can you have the comment address why this works in the non-recursive case as well? e.g. reused buffers are guaranteed to be cleared because the buffer is always cleared before it is returned to the pool.